src/charm.py

```diff
@@ -416,8 +428,14 @@ def _are_db_relations_ready(self) -> bool:
         if not self._stored.redis_relation:  # type: ignore
             self.model.unit.status = WaitingStatus("Waiting for redis relation")
             return False
-        if self._get_redis_relation_data()[0] == "":
-            self.model.unit.status = WaitingStatus("Waiting for redis relation to initialize")
+        try:
+            if (
+                self._get_redis_relation_data()[0] in ("", "None")
+                or self._get_redis_relation_data()[1] == 0
+            ):
```

Comment:
I'm not sure you need this now do you? You just need to run `_get_redis_relation_data()` and if it doesn't raise an error you know the data is good?

Reply:
If the redis port is None, the str(None) will not raise an exception. Same for a 0 port. Do we want to support port 0 ?

Reply:
Ok, this seems fine for now, and I think should later be refactored into a `redis`  module similar to the `database` module, but that can/should be handled separately (we should use a similar pattern for both).